CIP-0072/README.md

```diff
@@ -0,0 +1,614 @@
+```
+string = bstr .size (1..64) ; tstr / string from 1 up to 64 chars only
+
+sig_256 = bstr .size (64..64) ; 256 bytes signature (256 / 64 = 4 bytes)
+
+transaction_metadata = {
+  1667: on-chain_metadata
+}
+
+on-chain_metadata = {
+  subject: string,
+  rootHash: sig_256,
+  metadata: [+ string] / [+ string / [+ string]],
+  type: registration / de-registration,
+  signature: signature,
+}
+
+registration = {
+  action: "REGISTRATION",
+  ? comment: string, ; max 64 chars
+}
+
+de-registration = {
+  action: "DE_REGISTRATION",
+  ? comment: string, ; max 64 chars
+}
+
+signature = {
+  r: string,
+  s: string,
+  algo: text .regexp "Ed25519-EdDSA",
+  pub: string,
+}
+```
+
+which can be expressed using JSON schema.
+
+### dApp on-chain certificate JSON Schema
+```json
+{
+    "$schema":"https://json-schema.org/draft/2020-12/schema",
+    "$id":"https://example.com/dApp.schema.json",
+    "title": "Cardano dApp Claim",
+    "description": "Registration of Cardano dApp claim.",
+    "type":"object",
+    "properties":{
+       "subject":{
+          "type":"string",
+                "minLength": 1,
+                "maxLength": 64,
+                "pattern":"^[0-9a-fA-F]{1,64}$",
+                "description":"Identifier of the claim subject (dApp). A UTF-8 encoded string, must be max 64 chars. Typically it is randomly generated hash by the dApp developer."
+       },
+       "rootHash":{
+          "type":"string",
+                "minLength": 64,
+                "maxLength": 64,
+                "pattern":"^[0-9a-fA-F]{64}$",
+               "description":"blake2b-256 hash of the metadata describing the off-chain part of the dApp."
+       },
+       "metadata": {
+         "type": "array",
+         "description": "An array of valid URLs pointing to off-chain CIP-72 compatible metadata document. If an individual URL is longer than 64 characters, it must be expressed as an array of strings (where each string may contain at most 64 characters).",
+         "items": {
+           "anyOf": [{
+             "type": "string",
+               "minLength": 1,
+               "maxLength": 64
+             }, {
+             "type": "array",
+             "items": {
+               "type": "string",
+               "minLength": 1,
+               "maxLength": 64
+             }
+           }],
+           "examples": ["https://raw.githubusercontent.com/org/repo/offchain.json", ["https://raw.githubusercontent.com/long-org-name/", "long-repo-name/offchain.json"], "ipfs://QmbQDvKJeo2NgGcGdnUiUFibTzuKNK5Uij7jzmK8ZccmWp", ["ipfs://QmbQDvKJeo2NgGcGdnUiaAdADA", "UFibTzuKNKc0jA390alDAD5Uij7jzmK8ZccmWp"]]
+         }
+       },
+       "type":{
+          "type":"object",
+          "description":"Describes the releases, if they are new or an updates.",
+          "properties":{
+             "action":{
+               "type":"string",
+               "enum":["REGISTER", "DE_REGISTER"],
+               "pattern": "^(REGISTER|DE_REGISTER)$",
```

Comment:
If this is defined as enum, do we additionally need the regex?

Reply:
probably not, good point